src/main/java/uk/gov/hmcts/reform/sscs/service/EventService.java

```diff
@@ -0,0 +1,112 @@
+package uk.gov.hmcts.reform.sscs.service;
+
+import static uk.gov.hmcts.reform.sscs.model.NotificationEventType.CREATE_APPEAL_PDF;
+
+import java.net.URI;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsDocument;
+import uk.gov.hmcts.reform.sscs.idam.IdamService;
+import uk.gov.hmcts.reform.sscs.idam.IdamTokens;
+import uk.gov.hmcts.reform.sscs.model.NotificationEventType;
+
+@Service
+@Slf4j
+public class EventService {
+
+    private final SscsPdfService sscsPdfService;
+    private final RoboticsService roboticsService;
+    private final EvidenceManagementService evidenceManagementService;
+    private final EmailService emailService;
+    private final RegionalProcessingCenterService regionalProcessingCenterService;
+    private final IdamService idamService;
+
+    @Autowired
+    EventService(SscsPdfService sscsPdfService,
+                 RoboticsService roboticsService,
+                 RegionalProcessingCenterService regionalProcessingCenterService,
+                 IdamService idamService,
+                 EvidenceManagementService evidenceManagementService,
+                 EmailService emailService) {
+        this.sscsPdfService = sscsPdfService;
+        this.roboticsService = roboticsService;
+        this.regionalProcessingCenterService = regionalProcessingCenterService;
+        this.idamService = idamService;
+        this.evidenceManagementService = evidenceManagementService;
+        this.emailService = emailService;
+    }
+
+    public boolean handleEvent(NotificationEventType eventType, SscsCaseData caseData) {
+
+        if (CREATE_APPEAL_PDF == eventType) {
+            createAppealPdfAndSendToRobotics(caseData);
+            return true;
+        }
+
+        return false;
+    }
+
+    private void createAppealPdfAndSendToRobotics(SscsCaseData caseData) {
+
+        if (!hasDocument(caseData)) {
+
+            caseData.setEvidencePresent(hasEvidence(caseData));
+            String firstHalfOfPostcode = regionalProcessingCenterService.getFirstHalfOfPostcode(
+                    caseData.getAppeal().getAppellant().getAddress().getPostcode());
```

Comment:
The Appellant will not have an address if it's the same address as the appointee. 
cc. @arcotc

Reply:
It is now handled.